### PR TITLE
Player slots work again

### LINKED ISSUE #1580: Job/occupation player slots don't update at all
# Description
When you join it shows all roles are available, no matter if they have been taken already resulting in being able to have multiple captains on a station.

## Category
BUG

## Steps to Reproduce
Please enter the steps to reproduce the bug or behaviour:
1. Open 2 clients (build and editor)
2. Host with one and connect with the other
3. Join the station crew with both
4. Select a job on one client
5. Notice the job slot doesn't get updated on the other

### PR DESCRIPTION
**Pull request by Bod9001**
### Purpose
Fixes #1580

### Notes:

Tested with editor and client

### CODE REVIEW

File: UnityProject/Assets/Scripts/Lifecycle/PlayerSpawn.cs [hunk 1]
```diff
@@ -41,6 +41,9 @@ public static void ServerSpawnPlayer(JoinedViewer forViewer, Occupation occupati
 			{
 				SecurityRecordsManager.Instance.AddRecord(newPlayer.GetComponent<PlayerScript>(), occupation.JobType);
 			}
+			var Element = PlayerList.GetConnectedPlayer(newPlayer);
+			Element.Job = occupation.JobType;
+			PlayerList.Instance.AddOrUpdate(Element);
```
Comment (Jibbajabbafic) [inline UnityProject/Assets/Scripts/Lifecycle/PlayerSpawn.cs]: There's a better place to do this without needing the new function. I'll push a commit onto your branch.

**Review by Jibbajabbafic — CHANGES_REQUESTED** [review]

File: UnityProject/Assets/Scripts/Lifecycle/PlayerSpawn.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Lifecycle/PlayerSpawn.cs]: oh, yeah Keep the function though since that's useful I think

**Comment by Bod9001** [comment]
is basically Jibby's Work now : P

**Review by Jibbajabbafic — APPROVED** [review]

**Comment by Jibbajabbafic** [comment]
Now you're the reviewer 😝 . Can you double check it works and merge it if it's all good?

**Comment by Bod9001** [comment]
yeah will do

**Comment by Bod9001** [comment]
looks good tested with editor and 2 clients

**Pull request merged**